tests/handlers/test_sliding_sync.py

```diff
@@ -572,9 +573,23 @@ def test_combine_room_sync_config(
         self._assert_room_config_equal(combined_config, expected, "A into B")
 
 
-class GetRoomMembershipForUserAtToTokenTestCase(HomeserverTestCase):
+# FIXME: This can be removed once we bump `SCHEMA_COMPAT_VERSION` and run the
+# foreground update for
+# `sliding_sync_joined_rooms`/`sliding_sync_membership_snapshots` (tracked by
+# https://github.com/element-hq/synapse/issues/17623)
+@parameterized_class(
+    ("use_new_tables",),
+    [
+        (True,),
+        (False,),
+    ],
+    class_name_func=lambda cls,
+    num,
+    params_dict: f"{cls.__name__}_{'new' if params_dict['use_new_tables'] else 'fallback'}",
```

Comment:
wonder if we can convince the formatter to format this differently? This formatting is rather obfuscating
```suggestion
    class_name_func=(
        lambda cls, num, params_dict:
        f"{cls.__name__}_{'new' if params_dict['use_new_tables'] else 'fallback'}"
    ),
```

Reply:
While the formatter seems to be happy with the alternate formatting, running the tests is not.
I get what you're saying about the formatting being unclear, but these shouldn't stick around too much longer since we can remove the fallback functionality of sliding sync now.